pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -824,71 +830,103 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 		delete(nc.knownNodeSet, deleted[i].Name)
 	}
 
+	// Marking the pods not ready on a node requires looping over them and
+	// updating each pod's status one at a time. This is performed serially, and
+	// can take a while if we're processing each node serially as well. So we
+	// process them with bounded concurrency instead, since most of the time is
+	// spent waiting on io.
+	sem := semaphore.NewWeighted(scheduler.UpdateWorkerSize)
+
+	var zoneToNodeConditionsLock sync.Mutex
 	zoneToNodeConditions := map[string][]*v1.NodeCondition{}
 	for i := range nodes {
-		var gracePeriod time.Duration
-		var observedReadyCondition v1.NodeCondition
-		var currentReadyCondition *v1.NodeCondition
-		node := nodes[i].DeepCopy()
-		if err := wait.PollImmediate(retrySleepTime, retrySleepTime*scheduler.NodeHealthUpdateRetry, func() (bool, error) {
-			gracePeriod, observedReadyCondition, currentReadyCondition, err = nc.tryUpdateNodeHealth(ctx, node)
-			if err == nil {
-				return true, nil
-			}
-			name := node.Name
-			node, err = nc.kubeClient.CoreV1().Nodes().Get(ctx, name, metav1.GetOptions{})
-			if err != nil {
-				klog.Errorf("Failed while getting a Node to retry updating node health. Probably Node %s was deleted.", name)
-				return false, err
-			}
-			return false, nil
-		}); err != nil {
+		if err := sem.Acquire(ctx, 1); err != nil {
 			klog.Errorf("Update health of Node '%v' from Controller error: %v. "+
-				"Skipping - no pods will be evicted.", node.Name, err)
+				"Skipping - no pods will be evicted.", nodes[i].Name, err)
 			continue
 		}
 
-		// Some nodes may be excluded from disruption checking
-		if !isNodeExcludedFromDisruptionChecks(node) {
-			zoneToNodeConditions[nodetopology.GetZoneKey(node)] = append(zoneToNodeConditions[nodetopology.GetZoneKey(node)], currentReadyCondition)
-		}
-
-		if currentReadyCondition != nil {
-			pods, err := nc.getPodsAssignedToNode(node.Name)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
-				if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
-					// If error happened during node status transition (Ready -> NotReady)
-					// we need to mark node for retry to force MarkPodsNotReady execution
-					// in the next iteration.
-					nc.nodesToRetry.Store(node.Name, struct{}{})
+		node := nodes[i].DeepCopy()
+		go func() {
+			start := time.Now()
```

Comment:
We should use the controller [now](https://github.com/kubernetes/kubernetes/blob/00aae4c10c204d258e62a96f30970e9b91b29149/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L269) func instead of `time.Now()` to prevent skew across the cluster.

Reply:
Please fix